src/bootchooser.c

```diff
@@ -1654,6 +1718,30 @@ static gboolean custom_set_primary(RaucSlot *slot, GError **error)
 	return custom_backend_set("set-primary", slot->bootname, NULL, error);
 }
 
+/* Get current booted bootname */
+gchar *r_boot_get_bootname(RaucConfig *config, GError **error)
+{
+	GError *ierror = NULL;
+	gchar *res = NULL;
+
+	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
```

Comment:
Add a check for the config parameter.

Reply:
... and should return `NULL`.

Reply:
also, I think there are issues with error handling and `g_return_val_if_fail()` in general.

if think most of the code does not check if an error if set to propagate or prefix error, after the function ends with an error, because the function `g_return_val_if_fail(..., FALSE)` does not set the `GError`.

``` c
if (func(&ierror) == FALSE) {
    g_propagate_error(error, ierror);
    return FALSE;
}
```

[g_propagate_error(dest, src)[https://docs.gtk.org/glib/func.propagate_error.html] says *src must be non-NULL.*

Reply:
The `g_return_*` macros log a critical error, which we make always fatal by `g_log_set_always_fatal()` in `main()`.